server/spikeCalendar.py

```diff
@@ -0,0 +1,58 @@
+from server import app
+from bs4 import BeautifulSoup
+from flask import jsonify
+import re
+import requests
+import datetime
+
+BASE_URL = "https://spike.wharton.upenn.edu/calendar/index.cfm?format=list&date="
+
+def cleanSpikeTimeString(s):
+  regex = re.compile("1?[0-9]:[0-9][0-9] [AP]M")
+  all_times = regex.findall(s)
+  return all_times
+
+def events_on_date(d):
+  """Pulls the calendar from the Penn website and
+  filters out which events are 2 weeks away from date d.
+
+  :param d: date object that specifies the date
+  """
+  r = requests.get(BASE_URL + d.strftime('%Y-%m/%d')) 
+  soup = BeautifulSoup(r.text, "html5lib")
+  cal = soup.find("body").find(id="content").find(id="calendar_content").find(id="calendar_event_list")
```

Comment:
Does this need the first 2 `find`s? `id` is unique, so unless there's malformed HTML only the last one is needed